### PR TITLE
Housekeeping: Kusk Services (#956): Put all kusk k8s services in one service

### LINKED ISSUE #956: Housekeeping: Kusk Services
The [service](https://github.com/kubeshop/kusk-gateway/blob/main/config/manager/service.yaml#L31-L45) created with each kusk installation should be done programmatically  [here](https://github.com/kubeshop/kusk-gateway/blob/main/internal/envoy/auth/ext_authz_parser.go#L47-L48).

Challenge: Since `kusk-gateway-auth-service` is only used when Cloudentity is configured we need to find a way of knowing when to remove the service if unused. Reconciler doesn't have any knowledge of the previous state of the cluster.

### PR DESCRIPTION
**Pull request by jasmingacic**
Fixes #956

---

Housekeeping: Kusk Services (#956)

Issue and Discussion
====================

See https://github.com/kubeshop/kusk-gateway/issues/956.

Resolves kubeshop/kusk-gateway#956.

Summary
=======

Rename `kusk-gateway-manager-service` to `kusk-gateway-manager`.

Remove `kusk-gateway-validator-service` and `kusk-gateway-auth-service` since they are now part of `kusk-gateway-manager`.

Changes
=======

`internal/services/services.go`
-------------------------------

Introduce `AuthServiceURL` and `ValidatorURL`, so we don't have to repeat service URLs in several places.

`config/manager/manager.yaml`
-----------------------------

Remove `command` from `containers` as we have `ENTRYPOINT ["/manager"]` in `build/manager/Dockerfile`, i.e., the block
below is not needed:

```yaml
        - command:
            - /manager
```

`internal/envoy/auth/parser.go`
-------------------------------

Fix possible null pointer exception when dereferencing `auth.Custom.Host.Path`.

`examples/auth/oauth2/static-route/**`
-------------------------------------

Remove `StaticRoute` example as `paths` is no longer applicable.

`SKAFFOLD.md`
-------------

Remove an item from `TODO` list section as it is no longer applicable.

---

Signed-off-by: jasmingacic <jasmin.gacic@gmail.com>
Signed-off-by: Mohamed Bana <mohamed@bana.io>

### CODE REVIEW

**Comment by jasmingacic** [comment]
Let's leave this as draft for some time.

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -23,9 +23,7 @@ spec:
       securityContext:
         runAsNonRoot: true
       containers:
-        - command:
```
Comment (kylehodgetts) [inline config/manager/manager.yaml:26]: Why have we removed the command block?

File: internal/authz/authz.go [hunk 2]
```diff
@@ -5,44 +5,46 @@ import (
...
 	"github.com/kubeshop/kusk-gateway/internal/cloudentity"
 )
 
 type AuthorizationServer struct {
 	log logr.Logger
 }
 
-func (a *AuthorizationServer) check(w http.ResponseWriter, r *http.Request) {
-	url := r.Header.Get(cloudentity.HeaderAuthorizerURL)
+func (a *AuthorizationServer) check(writer http.ResponseWriter, request *http.Request) {
```
Comment (kylehodgetts) [inline internal/authz/authz.go:16]: Not sure changing `w` and `r` to `writer` and `request` is adding anything
`w` and `r` is idiomatic

**Review by kylehodgetts — CHANGES_REQUESTED** [review]

File: internal/authz/authz.go (hunk 2 above)
Comment (mbana) [reply internal/authz/authz.go:16]: the linter was complaininig.

File: config/manager/manager.yaml (hunk 1 above)
Comment (mbana) [reply config/manager/manager.yaml:26]: It's not needed.

`config/manager/manager.yaml`
-----------------------------

Remove `command` from `containers` as we have `ENTRYPOINT ["/manager"]` in `build/manager/Dockerfile`, i.e., the block
below is not needed:

```yaml
        - command:
            - /manager
```

**Review by mbana — APPROVED** [review]
I've tried Cloudentity with the `kusk cluster install` and it worked fine after the change, so it LGTM.

Would be good if other people tried it, so let's keep this open for now until it's got further testing.

File: internal/authz/authz.go (hunk 2 above)
Comment (kylehodgetts) [reply internal/authz/authz.go:16]: Really? What was it saying?

Comment (kylehodgetts) [reply internal/authz/authz.go:16]: Hmm okay fair enough

**Review by kylehodgetts — APPROVED** [review]

**Pull request merged**